compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1004,6 +1004,11 @@ class Definitions {
   def isBottomType(tp: Type): Boolean =
     tp.derivesFrom(NothingClass) || tp.derivesFrom(NullClass)
 
+  /** Is `tp` a subtype of `Nothing` or `Null`? Unlike `isBottomType`, this uses subtyping instead of inheritance. */
+  def isSubtypeOfBottom(tp: Type): Boolean = {
```

Comment:
Having two methods called "isBottomType" and "isSubtypeOfBottom" is confusing. I can send a follow-up PR where we'd rename them to make clear that the former uses inheritance and the latter subtyping:
  - isBottomClass => inheritsBottomClass
  - isBottomType => inheritsBottomType
  - isSubtypeOfBottom => unmodified

WDYT?

Reply:
Why do we need that method? It's always bad to have two ways to do things. That's why so far we used only inheritance and never subtyping.

Reply:
I thought this was needed to handle the case Paolo mentioned:
```
type X = Null
val Some(_) = (null: X)
```
But I got it wrong and the old `isBottomType` can handle it.
Removed this function.